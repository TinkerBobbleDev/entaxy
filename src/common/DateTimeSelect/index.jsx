import React from 'react'
import PropTypes from 'prop-types'
import DateFnsUtils from 'material-ui-pickers/utils/date-fns-utils'
import MuiPickersUtilsProvider from 'material-ui-pickers/utils/MuiPickersUtilsProvider'
import DateTimePicker from 'material-ui-pickers/DateTimePicker'

class DateTimeSelect extends React.Component {
  handleChange = (value) => {
    // this is going to call setFieldValue and manually update values.createdAt
    this.props.onChange(this.props.name, value)
  }

  render() {
    return (
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <DateTimePicker
          label={this.props.label}
          inputProps={{ 'aria-label': 'Date', required: true }}
          value={this.props.value}
          name={this.props.name}
          onChange={this.handleChange}
        />
      </MuiPickersUtilsProvider>
    )
  }
}

DateTimeSelect.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired
}

export default DateTimeSelect
