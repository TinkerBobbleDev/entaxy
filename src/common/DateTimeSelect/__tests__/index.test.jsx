import React from 'react'
import renderer from 'react-test-renderer'
import DateTimeSelect from '../'

describe('DateTimeSelect', () => {
  it('matches snapshot', () => {
    const mochOnChange = jest.fn()
    const component = renderer.create((
      <DateTimeSelect
        label="Date"
        name="createdAt"
        value={new Date('September 20th 11:36 a.m.')}
        onChange={mochOnChange}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
