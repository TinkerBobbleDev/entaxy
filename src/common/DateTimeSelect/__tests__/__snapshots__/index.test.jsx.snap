// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeSelect matches snapshot 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-3"
  onClick={[Function]}
  onError={undefined}
  onKeyPress={[Function]}
>
  <label
    className="MuiFormLabel-root-12 MuiFormLabel-error-15 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11 MuiInputLabel-shrink-10"
    data-shrink={true}
    htmlFor={undefined}
  >
    Date
  </label>
  <div
    className="MuiInput-root-17 MuiInput-error-22 MuiInput-formControl-18 MuiInput-underline-21 WithUtils-DateTextField--input-2"
  >
    <input
      aria-invalid={true}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-25"
      defaultValue={undefined}
      disabled={false}
      id={undefined}
      mask={null}
      name="createdAt"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder={undefined}
      readOnly={true}
      required={undefined}
      rows={undefined}
      type="text"
      value="Unknown"
    />
  </div>
  <p
    className="MuiFormHelperText-root-30 MuiFormHelperText-error-31"
    id={undefined}
  >
    Invalid Date Format
  </p>
</div>
`;
